tests/mbed_tools/test_cli.py

```diff
@@ -0,0 +1,6 @@
+from mbed_targets.mbed_tools.list_all import list_all
+from mbed_targets.mbed_tools.cli import cli
+
+
+def test_exposes_list_all_as_entry_point():
+    assert cli == list_all
```

Comment:
for tests, could we use unittest package and its helper method instead of assert?

Reply:
FWIW, we had an offline discussion and @acabarbaye main concern was wrt to `self.assert*` family producing better error messages. (correct me if I'm wrong Adrien)

I'm not sure I agree with this, they both seem to have their flaws. Personally, I'm not a fan of verbosity of `unitttest.TestCase`. I much prefer succinct, yet not limiting `pytest` style where all you need is a function, but can go class if you so desire.
If we were to go with unittest style (unless it's already decided on) then I wonder why we use `pytest` in the first place. ;)
It's also worth mentioning, that you need workarounds for certain `unittest`y things with pytest (fixtures, `tearDown` when using `--pdb`).

Reply:
On balance I would prefer to stick with `unitttest.TestCase` and just use pyTest as a runner.

Reply:
👍 I shall update the test to use `unittest.TestCase` 😄